launcher/src/trusted-extensions.ts

```diff
@@ -0,0 +1,68 @@
+/**********************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import { env } from 'process';
+import { ProductJSON } from './product-json';
+
+export class TrustedExtensions {
+  async configure(): Promise<void> {
+    console.log('# Configuring Trusted Extensions...');
+
+    if (!env.VSCODE_TRUSTED_EXTENSIONS) {
+      console.log('  > env.VSCODE_TRUSTED_EXTENSIONS is not defined, skip this step');
+      return;
+    }
+
+    try {
+      const extensions: string[] = [];
+
+      const tmp = env.VSCODE_TRUSTED_EXTENSIONS.split(',');
+      for (const e of tmp) {
+        if (e) {
+          extensions.push(e);
+        }
+      }
+
+      if (!extensions.length) {
+        console.log('  > env.VSCODE_TRUSTED_EXTENSIONS does not specify any extension');
```

Comment:
I think it makes sense to add the value of the variable to the log
Let's say another separator is applied, then:
- I agree that `extensions` array is empty
- but `env.VSCODE_TRUSTED_EXTENSIONS does not specify any extension` is NOT true
- so it makes sense to clarify it in logs, this way it would help to identify and resolve the problem quickly

Reply:
User specified `env.VSCODE_TRUSTED_EXTENSIONS` in his devfile.
Launcher says that `env.VSCODE_TRUSTED_EXTENSIONS` is wrong.
No need here to duplicate the value.

From the other side, if you have an idea, please share it.

Reply:
yes, `Launcher says that env.VSCODE_TRUSTED_EXTENSIONS is wrong.`
but the launcher says nothing about:
- what value of the variable the launcher has got
- what could be wrong
- what is expected

You use here `,` as a separator for the `VSCODE_TRUSTED_EXTENSIONS` env var, 
but, for example, `PATH` env variable uses `:` as a separator

<img width="383" alt="image" src="https://github.com/che-incubator/che-code/assets/5676062/ad31edb4-a045-4e78-8373-c5506997c7c6">

If a user uses `:` as a separator (just out of habit) for  the `VSCODE_TRUSTED_EXTENSIONS` then:
-  `VSCODE_TRUSTED_EXTENSIONS` is defined, it's not empty, it contains extensions
- but it doesn't work and there is nothing helpful in logs - isn't it?

My propose was just clarify it in logs - like - if the logic can not handle the variable - then it should be clear:
- env variable `VSCODE_TRUSTED_EXTENSIONS` has `some value here` => 
- using `,` as a separator to parse extensions => 
- can not parse extensions, so the value of `VSCODE_TRUSTED_EXTENSIONS` will be ignored => 
- the expected format of the `VSCODE_TRUSTED_EXTENSIONS` is: `some-value,another-value`

============
anyway I've approved your pull request yesterday, these changes are minor, it's up to you
so feel free to merge the pull request as soon as you believe it's good enough

Reply:
Have just played with it and would say, that the cheapest way in case of errors is to print the current value of `VSCODE_TRUSTED_EXTENSIONS` environment variable and explain that it should contain one or several extension identifiers divided by comma.

It's for the first step.
The next, I think we should document the functionality provided by launcher. Among them, ability to configure trusted extensions, location of Open VSX registry, webviews, etc.

Reply:
The output with one failed extension
```
# Configuring Trusted Extensions...
  > env.VSCODE_TRUSTED_EXTENSIONS is set to [redhat.yaml,redhat.openshift,red hat.java]
  > add redhat.yaml
  > add redhat.openshift
  > failure to add [red hat.java] because of wrong identifier
```

The output when the variable does not specify anything
```
# Configuring Trusted Extensions...
  > env.VSCODE_TRUSTED_EXTENSIONS is set to [,,,]
  > ERROR: The variable provided most likely has wrong format. It should specify one or more extensions separated by comma.
```